core/include/detray/propagator/propagator.hpp

```diff
@@ -162,7 +164,7 @@ struct propagator {
     /// @note If the return value of this function is true, another step can
     /// be taken afterwards.
     DETRAY_HOST_DEVICE bool propagate_step(
-        state &propagation,
+        state &propagation, bool is_init,
```

Comment:
Why do we need this? I thought the assumption here would be that the propagator would always be initialized at this point. Need that not be true?

Reply:
Yes, the propagation is in a valid state at this point, but this is a flag that means the navigator called ```init``` at some point during its update, which may happen at any point, e.g. if the cache is exhausted. In that case, the stepper currently resets the step size to the distance to the next surface given by the navigator instead of doing adaptive step size scaling. I thought I should keep that behavior intact for this PR

Reply:
Okay, not sure I fully understand but I trust your judgement.

Reply:
I am not entirely sure about this behavior either, but let's keep it as it was for this PR.